components/tailscale/providers.tf

```diff
@@ -0,0 +1,53 @@
+locals {
+  aws_account_id = module.account.outputs.full_account_map[var.stage]
+  assume_role    = format("arn:aws:iam::${local.aws_account_id}:role/${var.namespace}-${var.stage}-%s", var.run_in_cicd ? "terraform" : "admin")
+}
+
+variable "run_in_cicd" {
+  type        = bool
+  description = "If run in CI/CD platform, use terraform delegated role, otherwise use admin."
+  default     = true
+}
+
+variable "assume_role_override" {
+  type        = string
+  description = "Overrides the assumed role for the AWS provider. Useful for local testing and imports."
+  default     = null
+}
+
+variable "region" {
+  type        = string
+  description = "AWS region."
+}
+
+variable "tailnet" {
+  type        = string
+  description = "The Tailnet to perform actions in."
+}
+
+module "account" {
+  source  = "cloudposse/stack-config/yaml//modules/remote-state"
+  version = "0.22.3"
+
+  component = "account-map"
+  stage     = "root"
+
+  context = module.this.context
+}
+
+
+provider "aws" {
+  region = var.region
+
+  assume_role {
+    role_arn     = coalesce(var.assume_role_override, local.assume_role)
+    session_name = var.run_in_cicd ? "cicd-terraform" : "local-admin"
+    external_id  = "terraform"
+  }
+}
+
+provider "tailscale" {
+  tailnet             = var.tailnet
+  oauth_client_id     = local.secrets.tailscale_oauth_client_id
+  oauth_client_secret = local.secrets.tailscale_oauth_client_secret
+}
```

Comment:
same here, I feel the way we are providing creds to the provider to be a little prescriptive, maybe I'm missing the point here and we really WANT to be prescriptive about this sort of stuff, but if I was managing infra I probably would have my stuff set up in a way that these are fed in via environment variables.

here we're placing a boundary to adoption on someone already storing these things in SSM

Reply:
Yeah, it's an open question. 
I like the idea of having some kind of secrets mixin (or probably module), that provides us an elegant way to work primarily with SOPS (and support any other manager we need). If we unfollow Atmos' way - we need to come up with something new. I don't know if we ever would want to pass a secret value as a Terraform variable, since usually it's configured as fetching secret value from some storage, but I'm open to any good example to reconsider.